pkg/planner/core/indexmerge_path.go

```diff
@@ -1193,10 +1193,17 @@ func buildPartialPaths4MVIndex(
 	if virColID == -1 { // unexpected, no vir-col on this MVIndex
 		return nil, false, false, nil
 	}
-	if len(accessFilters) <= virColID { // no filter related to the vir-col, build a partial path directly.
-		partialPath, ok, err := buildPartialPath4MVIndex(sctx, accessFilters, idxCols, mvIndex, histColl)
-		return []*util.AccessPath{partialPath}, false, ok, err
+	if len(accessFilters) <= virColID {
```

Comment:
```
mysql> show create table t;
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                              |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` int DEFAULT NULL,
  `d` json DEFAULT NULL,
  UNIQUE KEY `iad` (`a`,(cast(json_extract(`d`,_utf8mb3'$.b') as signed array))),
  KEY `ua` (`a`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)

mysql> explain  analyze select a from t use index(ua);
+-----------------------------------------------------------------------------------------------------+
| EXPLAIN                                                                                             |
+-----------------------------------------------------------------------------------------------------+
| -> Covering index scan on t using ua  (cost=0.45 rows=2) (actual time=0.072..0.084 rows=2 loops=1)
 |
+-----------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)

mysql> explain  analyze select a from t use index(iad);
+-----------------------------------------------------------------------------------+
| EXPLAIN                                                                           |
+-----------------------------------------------------------------------------------+
| -> Table scan on t  (cost=0.45 rows=2) (actual time=0.086..0.100 rows=2 loops=1)
 |
+-----------------------------------------------------------------------------------+
1 row in set (0.01 sec)

mysql> explain  analyze select d->'$.b' from t use index(iad);
+-----------------------------------------------------------------------------------+
| EXPLAIN                                                                           |
+-----------------------------------------------------------------------------------+
| -> Table scan on t  (cost=0.45 rows=2) (actual time=0.068..0.082 rows=2 loops=1)
 |
+-----------------------------------------------------------------------------------+
1 row in set (0.01 sec)
```

seems MySQL does know itself have some rows missing for mv index value, and avoids using mv index to just fetch data for those only mv index columns.

so here is clear, avoid using mv index to fetch data when json column's is not covered by conditions.

Reply:
> avoid using mv index to fetch data when json column's is not covered by conditions

Yes. That's what I was trying to do in this PR :beers: .